### DISCUSSION TITLE
[ENHANCEMENT] Allow specifying a custom HttpClient to be used by MAUI

### DISCUSSION THREAD

**Discussion by ahmedalejo** [dotnet/maui#142] [General]
On Android AndroidPlatformServices used for loading streams from Uris uses a HttpClient which a new instance is created for every time a resource needed.

Allowing to specify a custom HttpClient would enable numerous benefits:
- Security: 
  - Allowing custom SSL certificate validation for media resources(which should be within the scope of MAUI)
  - Allowing proxy configuration
  - With a DelegatingHandler monitor every traffic from your application
  - Allowing proxy configuration
  - Authenticated access to remote resources for gif, video or any other media asset that is supported,
- Performance: 
  - Would allow caching of resources via a custom DelegatingHandler.
  - Url re-writes to local CDN for remote assets during development.

The list goes on, just think of the benefits of a DelegatingHandler and you can add it to the list.

This isn´t specific to Android and should be considered as a whole.

Hopefully there still time for a change in MAUI
## Reference
https://github.com/dotnet/maui/blob/85ee7fe836dc005d44eb0ba6b4660b1c6341d59c/System.Maui.Platform.Android/Forms.cs#L617
https://github.com/dotnet/maui/blob/85ee7fe836dc005d44eb0ba6b4660b1c6341d59c/System.Maui.Platform.Android/Forms.cs#L817-L827

**Comment by ahmedalejo** [comment]
Please kindly inform if this isn´t the kind of contribution that you (i.e. [#dotnet/maui](https://github.com/dotnet/maui)) find useful. Been using Xamarin.Forms exclusively for mobile dev.

MAUI concept has been treating several pain points existent in Xamarin.Forms. 
I´d like to help stair it along the way.

**Comment by davidbuckleyni** [comment]
This is not down to Android this is your approach in design you should be creating an http client using singleton approach this is not platform specific but the way you develop ur software

**Comment by ahmedalejo** [comment]
@davidbuckleyni to whom is the "**you**" in your comment referring to? (i clarified mine)

**Comment by dersia** [comment]
I am not sure about the implementation of HttpClient on Android (talking about the dotnet frameworks implementation, not the xamarin Android Platform Services implementation), but using the `using`-pattern with HttpClient on windows is very very very bad.
[As you can read here in full](https://www.nimaara.com/beware-of-the-net-httpclient/amp/)

It's in general a good idea to switch to `HttpClientFactory`
